### PR TITLE
Multitransfer api

### PR DESCRIPTION
**Pull request by sasurobert**
Integration of esdt multi transfer

### CODE REVIEW

File: arwen/contexts/output.go [hunk 1]
```diff
@@ -387,6 +376,39 @@ func (context *outputContext) TransferESDT(
...
+		outTransfer, ok := vmOutput.OutputAccounts[string(destination)]
+		if ok && len(outTransfer.OutputTransfers) == 1 {
+			return outTransfer.OutputTransfers[0].Data
+		}
+	}
+
+	if len(transfers) == 1 {
+		data := []byte(vmcommon.BuiltInFunctionESDTNFTTransfer + "@" + hex.EncodeToString(transfers[0].ESDTTokenName) +
+			"@" + hex.EncodeToString(big.NewInt(0).SetUint64(transfers[0].ESDTTokenNonce).Bytes()) + "@" + hex.EncodeToString(transfers[0].ESDTValue.Bytes()))
+		data = append(data, []byte("@"+hex.EncodeToString(destination))...)
```
Comment (camilbancioiu) [inline arwen/contexts/output.go:400]: Format as one `append` per line, for readability.

File: arwen/elrondapi/elrondei.go [hunk 2]
```diff
@@ -1142,6 +1146,68 @@ func v1_3_transferESDTNFTExecute(
...
+	functionLength int32,
+	numArguments int32,
+	argumentsLengthOffset int32,
+	dataOffset int32,
+) int32 {
+	host := arwen.GetVMHost(context)
+	runtime := host.Runtime()
+	metering := host.Metering()
+
+	callArgs, err := extractIndirectContractCallArgumentsWithoutValue(
```
Comment (camilbancioiu) [inline arwen/elrondapi/elrondei.go:1167]: Optionally return early if `numTokenTransfers == 0`, but consuming the gas cost of one ESDTTransfer.

**Review by camilbancioiu — CHANGES_REQUESTED** [review]
Only a single change, in `elrondei.go`.

File: arwen/elrondapi/bigIntOps.go [hunk 3]
```diff
@@ -374,8 +382,10 @@ func v1_3_bigIntGetESDTCallValue(context unsafe.Pointer, destination int32) {
 	metering.UseGas(gasToUse)
 
 	value := bigInt.GetOne(destination)
-	esdtTransfer := getFirstESDTTransferIfExist(runtime.GetVMInput())
-	value.Set(esdtTransfer.ESDTValue)
+	esdtTransfer := getESDTTransferFromInput(runtime.GetVMInput(), index)
+	if esdtTransfer != nil {
+		value.Set(esdtTransfer.ESDTValue)
+	}
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/elrondapi/bigIntOps.go:388]: good find.

File: arwen/elrondapi/elrondei.go (hunk 2 above)
Comment (sasurobert) [reply arwen/elrondapi/elrondei.go:1167]: done

File: arwen/contexts/output.go (hunk 1 above)
Comment (sasurobert) [reply arwen/contexts/output.go:400]: reformatted - append was not needed

**Review by camilbancioiu — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**